lib/flow_commerce/flow_api_v0_client.rb

```diff
@@ -22949,10 +22953,75 @@ def to_hash
       # ===== END OF SERVICE DEFINITION =====
       module HttpClient
 
+        class HttpHandler
+
+          attr_reader :base_uri
+        
+          def initialize(base_uri)
+            Preconditions.assert_class('base_uri', base_uri, String)
+            @base_uri = URI(base_uri)
+          end
+
+          # Executes a request. The provided request object will be an
+          # instance of Net::HTTP (e.g. Net::HTTP::Get)
+          def execute(request)
+            raise "Override in subclass"
+          end
+
+        end
+        
+        class DefaultHttpHandler < HttpHandler
+
+          attr_reader :client
+          
+          def initialize(base_uri)
+            super(base_uri)
+
+            @client = Net::HTTP.new(@base_uri.host, @base_uri.port)
+            if @base_uri.scheme == "https"
```

Comment:
If this is to be general enough for any apidoc client generated, i think this is fine. we can force the use of https in the code that uses this library